src/utils/gettext.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module typing/utils/gettext
+ */
+
+/**
+ * Returns the whole text from a given range by adding all data from the text nodes together.
+ *
+ * **Note** The text is trimmed to the last occurrence of any inline element (e.g. `<softBreak>`).
```

Comment:
I'd remove this note. `<softBreak>` is not a text node so it is already covered by the information from the first sentence of the description. Just don't forget to write that the returned range can be trimmed (to the first and to the last text node from the original range).